x-pack/plugins/ml/public/application/timeseriesexplorer/timeseriesexplorer_embeddable_chart/timeseriesexplorer_title.tsx

```diff
@@ -0,0 +1,152 @@
+  entity: MlEntity;
+  operation: MlEntityFieldOperation;
+}> = ({ api, entity, operation }) => {
+  const {
+    services: { uiActions },
+  } = useMlKibana();
+  const isAddFilter = operation === ML_ENTITY_FIELD_OPERATIONS.ADD;
+  return (
+    <EuiToolTip
+      content={
+        isAddFilter ? (
+          <FormattedMessage
+            id="xpack.ml.timeSeriesExplorer.title.addFilterTooltip"
+            defaultMessage="Filter for"
+          />
+        ) : (
+          <FormattedMessage
+            id="xpack.ml.timeSeriesExplorer.title.addNegateFilterTooltip"
+            defaultMessage="Filter out"
+          />
+        )
+      }
+    >
+      <EuiButtonIcon
+        size="xs"
+        onClick={() => {
+          const trigger = uiActions.getTrigger(SINGLE_METRIC_VIEWER_ENTITY_FIELD_SELECTION_TRIGGER);
+          trigger.exec({
+            embeddable: api,
+            data: [
+              {
+                ...entity,
+                operation: isAddFilter
+                  ? ML_ENTITY_FIELD_OPERATIONS.ADD
+                  : ML_ENTITY_FIELD_OPERATIONS.REMOVE,
+              },
+            ],
+          });
+        }}
+        iconType={isAddFilter ? 'plusInCircle' : 'minusInCircle'}
+        aria-label={
+          isAddFilter
+            ? i18n.translate('xpack.ml.timeSeriesExplorer.title.addFilterAriaLabel', {
+                defaultMessage: 'Filter for',
+              })
+            : i18n.translate('xpack.ml.timeSeriesExplorer.title.addNegateFilterAriaLabel', {
+                defaultMessage: 'Filter out',
+              })
+        }
+      />
+    </EuiToolTip>
+  );
+};
+
+interface SingleMetricViewerTitleProps {
+  api: any;
+  entityData: { entities: MlEntity[]; count: number };
+  functionLabel: string;
+}
+
+export const SingleMetricViewerTitle: FC<SingleMetricViewerTitleProps> = ({
+  api,
+  entityData,
+  functionLabel,
+}) => {
+  return (
+    <EuiFlexGroup direction="column" gutterSize="none">
+      <EuiFlexItem grow={false}>
+        <EuiFlexGroup gutterSize="none" alignItems="center">
+          <EuiFlexItem grow={false}>
+            <EuiTitle size={'xs'}>
+              <h2>
+                <span>
+                  {i18n.translate('xpack.ml.timeSeriesExplorer.singleTimeSeriesAnalysisTitle', {
+                    defaultMessage: 'Single time series analysis of {functionLabel}',
+                    values: { functionLabel },
+                  })}
+                </span>
+              </h2>
+            </EuiTitle>
+          </EuiFlexItem>
+          <EuiFlexItem grow={false}>
+            <TimeSeriesExplorerHelpPopover embeddableMode />
+          </EuiFlexItem>
+        </EuiFlexGroup>
+      </EuiFlexItem>
+      <EuiFlexItem grow={false}>
+        <EuiFlexGroup alignItems="center">
+          {entityData.entities.map((entity, i) => {
+            return (
+              <EuiFlexItem grow={false}>
+                <EuiFlexGroup gutterSize="none" alignItems="center">
```

Comment:
Is there any tweak we can make to the flex layout here to prevent the large gap between the entity field name / value text and the filter icons as the labels grow in length?

<img width="1259" alt="Screenshot 2024-11-07 at 13 09 30" src="https://github.com/user-attachments/assets/2e04f151-018b-4478-9060-a75a705e88b4">

For that job I used the `filebeat_ecs` QAF bootstrap template, and then this job config:

```
"analysis_config": {
    "bucket_span": "1h",
    "detectors": [
      {
        "detector_description": "mean(\"http.response.body.bytes\") by \"http.request.referrer\" partitionfield=\"host.id\"",
        "function": "mean",
        "field_name": "http.response.body.bytes",
        "by_field_name": "http.request.referrer",
        "partition_field_name": "host.id",
        "detector_index": 0
      }
    ],
    "influencers": [
      "host.id",
      "http.request.referrer"
    ],
    "model_prune_window": "30d"
  },
```

Reply:
Looks like this is part of the flex components responsive behavior - the text tries to wrap at the hyphens until it can fit fully 🤔 I tried using `EuiTextTruncate` but we end up losing some of that responsiveness. 

https://github.com/user-attachments/assets/9c976a6f-5055-4351-ac3f-a3e9b62eb35a